### PR TITLE
lua/filters: kayvee_influxdblinebatch

### PR DESCRIPTION
**Pull request by nathanleiby**
This adds support for the influxdb batching filter to read all routing information from the message fields themselves, instead of the setting it in a config object.

To configure new routing, you'll simply pass new information on the message itself, rather than redeploying Heka with an updated config.

These are the corresponding Influx changes that go with https://github.com/Clever/heka-clever-plugins/pull/63

### CODE REVIEW

**Comment by nathanleiby** [comment]
cc @xavi-

File: lua/filters/kayvee_influxdblinebatch_spec.lua [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+-- returns the number of lines in the msg "data"
+function line_count(msg)
+    -- there's always a trailing "\n" we don't want to count
+    return #util.lines(msg.data) - 1
+end
+
+describe("Kayvee Influxdbline Batch Filter", function()
+    -- Prep mocks, which are re-used in multiple tests
+    local mock_cfg = {
+		name = "name",
```
Comment (nathanleiby) [inline lua/filters/kayvee_influxdblinebatch_spec.lua]: going to figure out what happened with tabbing here..

Comment (nathanleiby) [inline lua/filters/kayvee_influxdblinebatch_spec.lua]: retab: 71d6e5b74ef127bbd9be7af48c5564ac4f49af51

File: lua/filters/field_util.lua [hunk 2]
```diff
@@ -0,0 +1,148 @@
+-- This Source Code Form is subject to the terms of the Mozilla Public
```
Comment (nathanleiby) [inline lua/filters/field_util.lua]: NOTE: this is copied from Heka, and intended to be used purely for testing

File: lua/filters/kayvee_influxdblinebatch.lua [hunk 3]
```diff
@@ -0,0 +1,303 @@
...
+
+- decimal_precision (string, optional, default "6")
+    String that is used in the string.format function to define the number
+    of digits printed after the decimal in number values.  The string formatting
+    of numbers is forced to print with floating points because InfluxDB will
+    reject values that change from integers to floats and vice-versa.  By
+    forcing all numbers to floats, we ensure that InfluxDB will always
+    accept our numerical values, regardless of the initial format.
+
+- msg_type (string, optional, defaults to "signalfxbatch")
```
Comment (burnsed) [inline lua/filters/kayvee_influxdblinebatch.lua]: `s/signlafx/influxdb/g`

File: lua/filters/kayvee_influxdblinebatch.lua [hunk 4]
```diff
@@ -0,0 +1,303 @@
...
+    if not fields or #fields == 0 then return nil end
+
+    if tags and #tags > 0 then
+        return string.format("%s,%s %s %d", escape_string(series), table.concat(tags, ","), table.concat(fields, ","), ts)
+    else
+        return string.format("%s %s %d", escape_string(series), table.concat(fields, ","), ts)
+    end
+end
+
+local api_messages = {}
```
Comment (burnsed) [inline lua/filters/kayvee_influxdblinebatch.lua]: group all local vars together at the top?

File: lua/filters/kayvee_influxdblinebatch.lua [hunk 5]
```diff
@@ -0,0 +1,303 @@
...
+    return dims
+end
+
+local function escape_string(str)
+    return tostring(str):gsub("([ ,])", "\\%1")
+end
+
+local function encode_scalar_value(value, decimal_precision)
+    if type(value) == "number" then
+        -- Always send numbers as formatted floats, so InfluxDB will accept
```
Comment (burnsed) [inline lua/filters/kayvee_influxdblinebatch.lua]: great comments!

File: lua/filters/kayvee_influxdblinebatch_spec.lua [hunk 6]
```diff
@@ -0,0 +1,146 @@
+-- Allow importing from parent directory
```
Comment (burnsed) [inline lua/filters/kayvee_influxdblinebatch_spec.lua]: woot! tests!

**Comment by burnsed** [comment]
Generally looks great. just a few nits

File: lua/filters/kayvee_influxdblinebatch.lua (hunk 4 above)
Comment (nathanleiby) [inline lua/filters/kayvee_influxdblinebatch.lua]: Moved locals to top 96b5ba7d9d70b2d9ad45c4c05b448add8a9321d8

File: lua/filters/kayvee_influxdblinebatch.lua (hunk 3 above)
Comment (nathanleiby) [inline lua/filters/kayvee_influxdblinebatch.lua]: fixed signalfx refs throughout 5ec69b91beb87339e6fa50ff4c9f81b2794d3227

File: lua/filters/kayvee_influxdblinebatch.lua (hunk 5 above)
Comment (nathanleiby) [inline lua/filters/kayvee_influxdblinebatch.lua]: @mohit is to thank! (this comes from the prior influxdblinebatch filter)

**Pull request merged**